pandas/core/tools/datetimes.py

```diff
@@ -361,7 +361,18 @@ def _convert_listlike_datetimes(
     # warn if passing timedelta64, raise for PeriodDtype
     # NB: this must come after unit transformation
     orig_arg = arg
-    arg, _ = maybe_convert_dtype(arg, copy=False)
+    try:
+        arg, _ = maybe_convert_dtype(arg, copy=False)
+    except TypeError:
+        if errors == "coerce":
```

Comment:
actually do we have coverage for all of these paths? i seem to recall we using similar code elsewhere?

Reply:
yes, we have tests in tests.tools.test_to_datetime that hit all of these cases

Reply:
kk

also maybe able to de-duplicate some of this code